app/hotels/src/singleHotel/summary/RoomSummary.js

```diff
@@ -2,86 +2,198 @@
+  Price,
+  BottomSheetHandle,
+  VerticalSwipeResponder,
+} from '@kiwicom/mobile-shared';
 import { defaultTokens } from '@kiwicom/mobile-orbit';
 import { View } from 'react-native';
 import { createFragmentContainer, graphql } from '@kiwicom/mobile-relay';
+import memoize from 'memoize-one';
 
 import SummaryRow from './SummaryRow';
 import type { RoomSummary_room as Room } from './__generated__/RoomSummary_room.graphql';
 import SummaryButtons from './SummaryButtons';
+import ExtraCharges from './ExtraCharges';
 
 type Props = {|
   +room: ?Room,
   +goBack: () => void,
 |};
 
-export class RoomSummary extends React.Component<Props> {
-  calculateSelectedData = () => {
-    const availableRooms = this.props.room?.availableRooms;
+type State = {|
+  +isExpanded: boolean,
+|};
 
-    if (!availableRooms) {
-      return null;
-    }
-    const positiveSelections = availableRooms.filter(room => {
-      const selectedCount = room?.selectedCount ?? 0;
-      return selectedCount > 0;
-    });
+const getSelectedRooms = memoize((props: Props) => {
+  const availableRooms = props.room?.availableRooms ?? [];
+  return availableRooms.filter(room => room?.selectedCount);
+});
+
+const getMaxNumberOfGuestsInSelection = memoize(selectedRooms => {
+  return selectedRooms.reduce((acc, room) => {
+    const selectedCount = room?.selectedCount ?? 0;
+    const max = room?.room?.maxPersons ?? 0;
+    return acc + max * selectedCount;
+  }, 0);
+});
+
+export class RoomSummary extends React.Component<Props, State> {
+  state = {
+    isExpanded: false,
+  };
+
+  calculateSelectedData = () => {
+    const positiveSelections = getSelectedRooms(this.props);
 
-    if (positiveSelections.length === 0) {
-      return null;
-    }
     const currency =
       positiveSelections[0]?.incrementalPriceWithExtraCharges?.[0]?.price
         ?.currency;
-    const amountAndPersons = positiveSelections.reduce(
+    // I wanted to split these reducers into separate functions
+    // However flow does not seem to think that it is a good idea 😞
+    return positiveSelections.reduce(
       (acc, room) => {
         const selectedCount = room?.selectedCount ?? 0;
         const incrementalPrice =
           room?.incrementalPriceWithExtraCharges?.[selectedCount - 1];
-        const amount = incrementalPrice?.price?.amount ?? 0;
-        const maxPersons = room?.room?.maxPersons ?? 0;
+        const extraCharges = incrementalPrice?.extraCharges ?? [];
+
+        const calculatedExtraCharges = extraCharges.reduce(
+          (acc, charge) => {
+            const amount: number = parseFloat(charge?.amount) ?? 0;
+            if (amount === 0) {
+              // Need more information about calculation
+              // We might need to know exactly how may persons are in each room
+              // Right now we don't, so we get a diff with what is in the booking.com webview
+              return acc;
+            }
+            const chargeAmount = charge?.chargeAmount ?? '';
```

Comment:
Maybe `chargeAmount` could be misleading, considering it refers here to percentages. But I guess for city tax and others, it could really be an amount.  Not sure what the best naming would be.

Reply:
Yeah, it could be either percent, or the citytax pr person pr night, or pr stay.